src/Cake.Scripting.Abstractions/Models/LineChange.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace Cake.Scripting.Abstractions.Models
+{
+    public sealed class LineChange
```

Comment:
Perhaps also implement `IEquatable<LineChange>` which is what we really want and avoid casting in `Equals(object obj)`.